src/public/js/util.js

```diff
@@ -185,13 +264,92 @@ var UTIL = (function() {
         throw new Error('Division did not happen properly');
       }
     }
-    return remainder;
+
+    return remainder; // trimLeadingZeros breaks this -- length expectations
+  }
+
+  // Returns evaluation in GF(256) of polynomial at value
+  // polynomial is represented as a big-endian array, IE: [1,2,3] -> x^2 + 2x + 3
+  // value is represented as an integer
+  function polynomialEval(polynomial, value) {
+    var valuePower = 1; // updated to represent value^i during loop
+    var result = 0;
+    var term;
+    var coefficient;
+    for (var i = 1; i <= polynomial.length; i++) {
+      coefficient = polynomial[polynomial.length - i];
+      term = fieldMult(coefficient, valuePower);
+      result = result ^ term; // Galois extension field addition is xor with base field 2
+      valuePower = fieldMult(valuePower, value);
+    }
+    return result;
+  }
+
+  // Returns product of polynomial and scalar in GF(256)
+  // polynomial is represented as a big-endian array, IE: [1,2,3] -> x^2 + 2x + 3
+  // value is represented as an integer
+  function polynomialScale(polynomial, scalar) {
+    if (scalar == 0) {
+      return new Uint8Array(0);
+    }
+    polynomial = trimLeadingZeros(polynomial);
```

Comment:
instead of creating a new array to return use the array returned by trimLeadingZeros(polynomial) and simply replace the value at each index as you scale it. This would be 1 less array copy made.